### PR TITLE
Wip/hughsie/verfmt in history

### PR DESCRIPTION
**Pull request by hughsie**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [ ] Code fix
- [X] Feature
- [ ] Documentation

### CODE REVIEW

**Review by superm1 — COMMENTED** [review]
I think the plugin example should be updated too.

File: src/fu-self-test.c [hunk 1]
```diff
@@ -2532,6 +2533,42 @@ fu_engine_history_func(gconstpointer user_data)
...
+	fu_device_set_plugin(device, "test");
+	fu_device_add_guid(device, "12345678-1234-1234-1234-123456789012");
+	fu_device_add_checksum(device, "0123456789abcdef0123456789abcdef01234567");
+	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
+	fu_device_add_flag(device, FWUPD_DEVICE_FLAG_UNSIGNED_PAYLOAD);
+	fu_device_set_created(device, 1515338000);
+	fu_engine_add_device(engine, device);
+	g_assert_cmpint(fu_device_get_version_format(device), ==, FWUPD_VERSION_FORMAT_TRIPLET);
+	g_assert_cmpstr(fu_device_get_version(device), ==, "0.1.27");
+}
```
Comment (superm1) [inline src/fu-self-test.c:2570]: Probably a crazy scenario; but I think it's worth defining at least in the tests what happens if the version format changes from one daemon version to another.  For example a "Whoops, we meant that not to be raw but instead to be hex!".

File: libfwupdplugin/README.md [hunk 2]
```diff
@@ -120,10 +120,10 @@ Remember: Plugins should be upstream!
...
 * `fu_udev_device_get_fd()`: Use `fu_udev_device_get_io_channel()` instead
 * `fu_device_emit_request()`: Add `FuProgress` and `GError`
-* `fu_device_set_version_from_uint16()`: Use `fu_device_set_version_u16()` instead
-* `fu_device_set_version_from_uint24()`: Use `fu_device_set_version_u24()` instead
-* `fu_device_set_version_from_uint32()`: Use `fu_device_set_version_u32()` instead
-* `fu_device_set_version_from_uint64()`: Use `fu_device_set_version_u64()` instead
+* `fu_device_set_version_from_uint16()`: Use `fu_version_from_uint16()` instead
+* `fu_device_set_version_from_uint24()`: Use `fu_version_from_uint24()` instead
+* `fu_device_set_version_from_uint32()`: Use `fu_version_from_uint32()` instead
+* `fu_device_set_version_from_uint64()`: Use `fu_version_from_uint64()` instead
```
[the remark this answers is not in the document]
Comment (hughsie) [reply libfwupdplugin/README.md:126]: Ack, fixed.

**Comment by hughsie** [comment]
> I think the plugin example should be updated too.

Added

File: src/fu-self-test.c (hunk 1 above)
Comment (hughsie) [reply src/fu-self-test.c:2570]: I actually tested that -- which is why we only set a history-derived version format if the current device version format is UNKNOWN.

**Review by superm1 — APPROVED** [review]

**Pull request merged**